File: 1-Easy/Temperatures/temperatures.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int closest = 0;

        if (n != 0)
        {
            IEnumerable<int> inputs = Console.ReadLine().Split(' ').Select(int.Parse);

            // sorts in ascending order by the absolute values of the numbers first
            // then sorts by descending order so this way positives come before negatives
            closest = inputs.OrderBy(Math.Abs)
                            .ThenByDescending(x => x)
                            .FirstOrDefault();
        }
        Console.WriteLine(closest);
    }
}

File: 1-Easy/Temperatures/temperatures.js
const n = parseInt(readline());
let closest = 0;

if (n !== 0) {
    const inputs = readline().split(' ').map(Number);

    // sorts in ascending order if the absolute values of the numbers does not equal 0
    // otherwise, sorts in descending order, this way positives come before negatives
    inputs.sort((a, b) => Math.abs(a) - Math.abs(b) || b - a);
    closest = inputs[0];
}
console.log(closest);